vendor/github.com/tudalex/seccomp-sandbox/sandbox.go

```diff
@@ -0,0 +1,98 @@
+struct sigaction old_action;
+void handler(int signum, siginfo_t *info, void *context) {
+	//TODO: Put this in a nice formatted json error
+    fprintf(stderr,"Jail violation caused by syscall %s. Code %d\n", syscall_mappings[info->si_syscall], info->si_syscall);
+    //fprintf(stderr,"Code %d\n", info->si_syscall);
+    exit(1);
+}
+
+void install_sighandler() {
+    struct sigaction action;
+    sigaction(SIGSYS, NULL, &action);
+    memset(&action, 0, sizeof action);
+    sigfillset(&action.sa_mask);
+    action.sa_handler = NULL;
+    action.sa_sigaction = handler;
+    action.sa_flags = SA_NOCLDSTOP | SA_SIGINFO | SA_ONSTACK;
+    sigaction(SIGSYS, &action, &old_action);
+}
+*/
+import "C"
+
+import (
+	"github.com/seccomp/libseccomp-golang"
+	"log"
+)
+
+type FilterAction string
+
+type FilterOperation struct {
+	FilterOn   []string
+	Action     seccomp.ScmpAction
+	Conditions []seccomp.ScmpCondition
+}
+
+type SandboxProfile struct {
+	DefaultPolicy seccomp.ScmpAction
+	Filters       []FilterOperation
+}
+
+func Jail(sandboxProfile SandboxProfile) {
+	C.install_sighandler()
+	filter, err := seccomp.NewFilter(sandboxProfile.DefaultPolicy)
+	if err != nil {
+		log.Fatal("Error creating filter: %s\n", err)
+	} else {
+		log.Printf("Created filter\n")
+	}
+	defer filter.Release()
+	action, err := filter.GetDefaultAction()
+	if err != nil {
+		log.Fatal("Error getting default action of filter\n")
+	} else if action != seccomp.ActTrap {
+		log.Printf("Default action of filter was set incorrectly!\n")
+	}
+	log.Printf("%s\n", sandboxProfile.DefaultPolicy)
+	for _, profileFilter := range sandboxProfile.Filters {
+		for _, callName := range profileFilter.FilterOn {
+			call, err := seccomp.GetSyscallFromName(callName)
+			if err != nil {
+				log.Fatal("Error getting syscall number of %s: %s\n", callName, err)
+			} else {
+				log.Printf("Got hook to syscall %d\n", call)
+			}
+			// if there are conditions, construct a conditional rule
+			if len(profileFilter.Conditions) > 0 {
+				err = filter.AddRuleConditional(call, profileFilter.Action, profileFilter.Conditions)
+				if err != nil {
+					log.Fatal("Error adding conditional rule: %s", err)
+				}
+			} else {
+				err = filter.AddRule(call, profileFilter.Action)
+				if err != nil {
+					log.Fatal("Error adding rule to restrict syscall: %s\n", err)
+				} else {
+					log.Printf("Added rule to restrict syscall %s\n", callName)
+				}
+			}
+		}
+	}
+	filter.SetTsync(true)
+	filter.SetNoNewPrivsBit(true)
+	err = filter.Load()
```

Comment:
potentially, we could add code to detect if seccomp is supported here, for ex:
if it return errBadFilter then we have a filter error, log that the filter format appears incorrect/syntax error, fail and exit here (as in mig terminates)

Otherwise, check the syscall return, it should return unsupported or permission denied when seccomp is not supported, which we could log and continue, such as:

log.Printf("Failed to load seccomp filter, seccomp (filter mode) not supported by kernel: %s. Sandbox features disabled.", err)

See also: https://github.com/seccomp/libseccomp-golang/blob/master/seccomp.go#L592